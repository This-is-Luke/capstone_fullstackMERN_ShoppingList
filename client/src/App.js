import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';

import Home from './components/Home';
import Register from './components/Register';
import LoginThankyou from './components/LoginThankyou';
import Login from './components/Login';
import AdminHome from './components/AdminHome';
import AdminViewAllUsers from './components/AdminViewAllUsers';
import AdminViewUserList from './components/AdminViewUserList';
import AdminAddItems from './components/AdminAddItems';
import AdminEditList from './components/AdminEditList';
import UserHome from './components/UserHome';

function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/register" element={<Register />} />
        <Route path="/login-thankyou" element={<LoginThankyou />} />
        <Route path="/login" element={<Login />} />
        <Route path="/admin-home" element={<AdminHome />} />
        <Route path="/user-home" element={<UserHome />} />
        <Route path="/admin-view-all-users" element={<AdminViewAllUsers />} />
        <Route path="/admin-view-user-list" element={<AdminViewUserList />} />
        <Route path="/admin-add-items" element={<AdminAddItems />} />
        <Route path="/admin-edit-list" element={<AdminEditList />} />
      </Routes>
    </Router>
  );
}

export default App;
