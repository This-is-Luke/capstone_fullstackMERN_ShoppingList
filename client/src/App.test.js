import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import App from './App';
import Login from './components/Login';
import { useNavigate } from 'react-router-dom';

// Existing test
test('App renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Mock the useNavigate function from react-router-dom
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

// New test for Register button
test('Register button redirects to /register', () => {
  // Create a mock function for the navigation
  const navigate = jest.fn();
  
  // When useNavigate is called, return the mock navigation function
  useNavigate.mockReturnValue(navigate);

  // Render the Login component
  const { getByText } = render(<Login />);

  // Get the Register button
  const registerButton = getByText('Register');

  // Simulate a click event on the Register button
  fireEvent.click(registerButton);

  // Expect that the navigate function has been called with the /register route
  expect(navigate).toHaveBeenCalledWith('/register');
});
