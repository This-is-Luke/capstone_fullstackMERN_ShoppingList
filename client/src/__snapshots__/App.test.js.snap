// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="sc-bdDrbm dUqtr"
>
  <h1
    className="sc-hKVryf blHKXT"
  >
    Shopping List
  </h1>
  <p
    className="sc-eCQiwC hGNfhM"
  >
    <a
      className="sc-gtcAbF hCjGGE"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
     or
     
    <a
      className="sc-dkXrac bzcfDv"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </p>
</div>
`;
